server/src/modules/user/user.service.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  HttpException,
+  HttpStatus,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { PrismaClient } from '@prisma/client';
+import { UserDTO } from './user.dto';
+import * as bcrypt from 'bcrypt';
+
+@Injectable()
+export class UserService {
+  constructor(private prisma: PrismaClient) {}
+
+  async hashPassword(password: string) {
+    return await bcrypt.hash(password, 10);
+  }
+
+  async comparePassword(password: string, hash: string) {
+    return await bcrypt.compare(password, hash);
+  }
+
+  async getUserByEmail(email: string, password: string) {
+    const user = await this.prisma.user.findUnique({
+      where: {
+        email,
+      },
+    });
+
+    if (!user) {
+      throw new NotFoundException('Email not found');
+    }
+
+    const isPasswordValid = await this.comparePassword(password, user.password);
+    if (!isPasswordValid) {
+      throw new HttpException('Invalid password', HttpStatus.UNAUTHORIZED);
+    }
+
+    return user;
+  }
+
+  async createUser(data: UserDTO) {
+    const user = await this.prisma.user.findUnique({
+      where: {
+        email: data.email,
+      },
+    });
+    if (data.email) {
+      if (user) {
+        throw new HttpException('Email already exists', HttpStatus.BAD_REQUEST);
+      }
+    }
```

Comment:
**Suggestion:** The `createUser` method should validate the email format before attempting to create a new user. Add email format validation to ensure that only valid email addresses are accepted. [data validation, importance: 7]
```suggestion
  async createUser(data: UserDTO) {
    if (!this.isValidEmail(data.email)) {
      throw new HttpException('Invalid email format', HttpStatus.BAD_REQUEST);
    }
    const user = await this.prisma.user.findUnique({
      where: {
        email: data.email,
      },
    });
    if (user) {
      throw new HttpException('Email already exists', HttpStatus.BAD_REQUEST);
    }
```